api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -356,6 +365,18 @@ func (m *AzureManagedMachinePool) validateEnableNodePublicIP() error {
 	return nil
 }
 
+func (m *AzureManagedMachinePool) validateSubnetName() error {
+	if m.Spec.SubnetName != nil {
+		subnetRegex := "^[a-zA-Z0-9][a-zA-Z0-9-]{0,78}[a-zA-Z0-9]$"
```

Comment:
tbh i gave chatgpt the docs and asked it to generate a regex :). I then added the test-cases accordingly to check if the regex was correct turned out chatgpt did a solid work. The max character length for a name is 80 so on safe side i guess it stuck to 0-78 :)

Reply:
> chatgpt
/honk

Reply:
> The max character length for a name is 80 so on safe side i guess it stuck to 0-78 :)

Let's make it 80 char length in that case! Gotta be consistent with the supported length to avoid edge cases, especially in CI scenarios.

Reply:
ack

Reply:
@nawazkh the regex generated is correct and ensures the name length is restricted till 80 characters. I added a +ve and -ve test case for the same please check.